Menu/MenuManager.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\BlockBundle\Menu;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ */
+final class MenuManager implements MenuManagerInterface
+{
+    /**
+     * @var MenuBuilderInterface[]
+     */
+    private $menus = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add(MenuBuilderInterface $menu)
+    {
+        $method = $this->getMenuMethod($menu);
+
+        $this->menus[$method] = $menu;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMenuMethods()
+    {
+        return array_keys($this->menus);
+    }
+
+    /**
+     * Returns the menu method name.
+     *
+     * @param MenuBuilderInterface $menu
+     *
+     * @return string
+     */
+    private function getMenuMethod(MenuBuilderInterface $menu)
+    {
+        $reflector = new \ReflectionClass($menu);
+        $namespace = $reflector->getNamespaceName();
+        $class = $reflector->getName();
+
+        $bundle = str_replace('\\', '', preg_replace('/(.*Bundle)\\\\.*/i', '$1', $namespace));
+
+        return $bundle.':'.$class.':create';
```

Comment:
There is a similar notation for referencing Twig templates, and it has been deprecated recently because "no one understands it". Maybe find something less magic?